src/parquet/file/metadata-internal.cc

```diff
@@ -0,0 +1,500 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <vector>
+
+#include "parquet/file/metadata.h"
+#include "parquet/schema/converter.h"
+#include "parquet/thrift/util.h"
+
+namespace parquet {
+
+// MetaData Accessor
+// Column metadata
+class ColumnMetaData::ColumnMetaDataImpl {
+ public:
+  explicit ColumnMetaDataImpl(const format::ColumnChunk* column) : column_(column) {}
+  ~ColumnMetaDataImpl() {}
+
+  // column chunk
+  int64_t file_offset() const { return column_->file_offset; }
+  const std::string& file_path() const { return column_->file_path; }
+
+  // column metadata
+  Type::type type() { return FromThrift(column_->meta_data.type); }
+
+  int64_t num_values() const { return column_->meta_data.num_values; }
+  std::shared_ptr<schema::ColumnPath> path_in_schema() {
+    return std::make_shared<schema::ColumnPath>(column_->meta_data.path_in_schema);
+  }
+
+  bool is_stats_set() const { return column_->meta_data.__isset.statistics; }
+
+  ColumnStatistics Stats() const {
+    const format::ColumnMetaData& meta_data = column_->meta_data;
+
+    ColumnStatistics result;
+    result.null_count = meta_data.statistics.null_count;
+    result.distinct_count = meta_data.statistics.distinct_count;
+    result.max = &meta_data.statistics.max;
```

Comment:
Not sure if it was a good choice to store pointers to strings in the `ColumnStatistics`. I would actually expect that `ColumnStatistics` at least own a reference (or the value) of min & max.